test/cpp_extensions/open_registration_extension.cpp

```diff
@@ -632,17 +632,12 @@ const at::Generator& default_generator(c10::DeviceIndex device_index) {
 }
 
 void fallback_with_undefined_tensor() {
-  at::Tensor first = at::empty((2,3)).to(at::DeviceType::PrivateUse1);
-  at::Tensor second = at::Tensor();
-  at::Tensor step = at::empty({}).fill_(2).to(at::DeviceType::PrivateUse1);
-  at::Tensor grad_scale = at::empty({}).fill_(0.00001).to(at::DeviceType::PrivateUse1);
-  at::Tensor found_inf = at::empty({}).fill_(1).to(at::DeviceType::PrivateUse1);
-  at::TensorList tensors = {first, first};
-  at::TensorList undefined_tensors = {first, second};
-  at::TensorList steps = {step, step};
-  return at::_fused_adamw_(tensors, tensors, tensors, tensors, undefined_tensors,
```

Comment:
so calling `_fused_adamw_` here with undefined inputs feels like it was just wrong (we now error because you are passing in a mutable, undefined tensor, which autograd is not allowed to bump the VC on).

To make the test a bit more realistic I updated it to use aten::index(). which accepts undefined tensor indices (which are not mutated)